persist/fs/commitlog/iterator.go

```diff
@@ -62,15 +64,29 @@ type iteratorRead struct {
 	annotation []byte
 }
 
+// ReadCommitLogPredicate is a predicate that allows the caller to determine
+// which commitlogs the iterator should read from
+type ReadCommitLogPredicate func(commitLogTime time.Time) bool
+
+// ReadAllCommitLogs can be passed as the ReadCommitLogPredicate for callers
+// that want a convenient way to read all the commitlogs
+func ReadAllCommitLogs(commitLogTime time.Time) bool { return true }
```

Comment:
Perhaps we can add a more readable option:
```go
func ReadAllPredicate() ReadEntryPredicate {
  return func(entryTime time.Time) bool { return true }
}
```

That way the calling code looks more readable and its more clear looking at the godocs how to get a type that matches `ReadCommitLogPredicate` easily, for instance `ReadAllCommitLogs` just looking at godoc is not clear it can be used as a predicate.  Also trying to avoid including the package name in exported types is a Good Thing(tm). e.g.:
```go
iter, err := commitlog.NewIterator(opts, commitlog.ReadAllPredicate())
```

Rather than current:
```go
iter, err := commitlog.NewIterator(opts, commitlog.ReadAllCommitLogs)
```